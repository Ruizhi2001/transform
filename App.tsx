import { NavigationContainer } from "@react-navigation/native";
import { createNativeStackNavigator } from "@react-navigation/native-stack";
import { Provider } from "react-redux";
import { store } from "./redux/store";
import { ContentScreen } from "./screens/ContentScreen";
import { ExercisingScreen } from "./screens/Fitness/ExercisingScreen";
import { FitnessHomeScreen } from "./screens/Fitness/FitnessHomeScreen";
import { RestScreen } from "./screens/Fitness/RestScreen";
import { WorkoutScreen } from "./screens/Fitness/WorkoutScreen";
import { LoginScreen } from "./screens/authentication/LoginScreen";
import { RegisterScreen } from "./screens/authentication/RegisterScreen";

const Stack = createNativeStackNavigator();

export default function App() {
  return (
    <NavigationContainer>
      <Provider store={store}>
        <Stack.Navigator screenOptions={{ headerShown: false }}>
          <Stack.Screen name="Login" component={LoginScreen} />
          <Stack.Screen name="Register" component={RegisterScreen} />
          <Stack.Screen name="Content" component={ContentScreen} />
          <Stack.Screen name="FitnessHome" component={FitnessHomeScreen} />
          <Stack.Screen name="Workout" component={WorkoutScreen} />
          <Stack.Screen name="Exercising" component={ExercisingScreen} />
          <Stack.Screen name="Rest" component={RestScreen} />
        </Stack.Navigator>
      </Provider>
    </NavigationContainer>
  );
}
