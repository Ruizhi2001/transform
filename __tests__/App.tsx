import React from "react";
import renderer, { ReactTestRendererJSON } from "react-test-renderer";
import App from "../App";

describe("<App />", () => {
  it("has 1 child", () => {
    const tree = renderer.create(<App />).toJSON();

    // Check if tree is an array (multiple children)
    if (Array.isArray(tree)) {
      expect(tree.length).toBe(1);
    }
    // Check if tree is a single JSON object (one child)
    else if (tree as ReactTestRendererJSON) {
      expect(tree?.children).toHaveLength(1);
    } else {
      // Handle other cases if needed
      throw new Error("Unexpected tree structure");
    }
  });

  it("matches the snapshot", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
