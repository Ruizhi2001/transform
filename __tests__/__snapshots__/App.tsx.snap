// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches the snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={true}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="Login"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <RCTSafeAreaView
            style={
              {
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                {
                  "paddingHorizontal": 25,
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#333",
                    "fontSize": 28,
                    "fontWeight": "500",
                    "marginBottom": 30,
                  }
                }
              >
                Login Page
              </Text>
              <View
                style={
                  {
                    "borderBottomColor": "#ccc",
                    "borderBottomWidth": 1,
                    "flexDirection": "row",
                    "marginBottom": 25,
                    "paddingBottom": 8,
                  }
                }
              >
                <TextInput
                  keyboardType="email-address"
                  onChangeText={[Function]}
                  placeholder="Email"
                  style={
                    {
                      "flex": 1,
                      "paddingVertical": 0,
                    }
                  }
                />
              </View>
              <View
                style={
                  {
                    "borderBottomColor": "#ccc",
                    "borderBottomWidth": 1,
                    "flexDirection": "row",
                    "marginBottom": 25,
                    "paddingBottom": 8,
                  }
                }
              >
                <TextInput
                  onChangeText={[Function]}
                  placeholder="Password"
                  secureTextEntry={true}
                  style={
                    {
                      "flex": 1,
                      "paddingVertical": 0,
                    }
                  }
                />
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "opacity": 1,
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "color": "#6699CC",
                        "fontWeight": "700",
                      }
                    }
                  >
                    Forgot?
                  </Text>
                </View>
              </View>
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "backgroundColor": "#6699CC",
                    "borderRadius": 10,
                    "marginBottom": 30,
                    "opacity": 1,
                    "padding": 20,
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "#fff",
                      "fontSize": 16,
                      "fontWeight": "700",
                      "textAlign": "center",
                    }
                  }
                >
                  Login
                </Text>
              </View>
            </View>
            <View
              style={
                {
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginBottom": 30,
                }
              }
            >
              <Text>
                New to the app?
              </Text>
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "#6699CC",
                      "fontWeight": "700",
                    }
                  }
                >
                   Register
                </Text>
              </View>
            </View>
          </RCTSafeAreaView>
        </View>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
